src/NHibernate.Test/NHSpecificTest/NH3848/TestFixture.cs

```diff
@@ -0,0 +1,266 @@
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.NH3848
+{
+	public abstract class TestFixture : TestCaseMappingByCode
+	{
+		protected Customer Customer1;
+		protected Customer Customer2;
+		protected Customer Customer3;
+		protected const int OrderNumber = 2;
+
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<Customer>(rc =>
+			{
+				rc.Table("Customers");
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.Name);
+				rc.Set(x => x.Orders, m =>
+				{
+					m.Inverse(true);
+					m.Key(k =>
+					{
+						k.Column("CustomerId");
+						k.NotNullable(true);
+					});
+					m.Cascade(Mapping.ByCode.Cascade.All.Include(Mapping.ByCode.Cascade.DeleteOrphans));
+					m.Cache(c => c.Usage(CacheUsage.ReadWrite));
+				}, m => m.OneToMany());
+			});
+
+			mapper.Class<Order>(rc =>
+			{
+				rc.Table("Orders");
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.Number);
+				rc.ManyToOne(x => x.Customer, m => m.Column("CustomerId"));
+				rc.Cache(c => c.Usage(CacheUsage.ReadWrite));
+			});
+
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		protected override void Configure(Configuration configuration)
+		{
+			base.Configure(configuration);
+			configuration.Cache(c =>
+			{
+				c.UseQueryCache = true;
+				c.Provider<HashtableCacheProvider>();
+			});
+		}
+
+		protected override void OnSetUp()
+		{
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				Customer1 = new Customer { Name = "First Customer" };
+
+				Customer2 = new Customer { Name = "Second Customer" };
+
+				Customer3 = new Customer { Name = "Third Customer" };
+
+				var customer1Order1 = new Order { Number = 1 };
+				Customer1.AddOrder(customer1Order1);
+
+				var customer1Order2 = new Order { Number = 2 };
+				Customer1.AddOrder(customer1Order2);
+
+				var customer2Order1 = new Order { Number = 1 };
+				Customer2.AddOrder(customer2Order1);
+
+				var customer2Order2 = new Order { Number = 2 };
+				Customer2.AddOrder(customer2Order2);
+
+				var customer3Order1 = new Order { Number = 1 };
+				Customer3.AddOrder(customer3Order1);
+
+				session.Save(Customer1);
+				session.Save(Customer2);
+				session.Save(Customer3);
+
+				transaction.Commit();
+				session.Flush();
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			ClearSecondLevelCacheFor(typeof(Customer));
+			ClearCollectionCache<Customer>(n => n.Orders);
+			ClearSecondLevelCacheFor(typeof(Order));
+
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public virtual void ChildCollectionsFromLeftOuterJoinWithOnClauseRestrictionOnCollectionShouldNotBeInSecondLevelCache()
+		{
+			var firstSession = OpenSession();
```

Comment:
I have left this pattern go into master for the `Fixture`, I shouldn't had. With some overrides throwing an ignore in this PR, this causes sessions to remain opened, triggering a test clean-up failure. `using` needs to be used.